docker.go

```diff
@@ -119,8 +124,51 @@ type (
 	TagStruct struct {
 		Tag string `json:"Tag"`
 	}
+
+	LayerStatus struct {
+		Status string
+		Time   float64 // Time in seconds; only set for DONE layers
+	}
+
+	CacheMetrics struct {
+		TotalLayers int                 `json:"total_layers"`
+		Done        int                 `json:"done"`
+		Cached      int                 `json:"cached"`
+		Errored     int                 `json:"errored"`
+		Cancelled   int                 `json:"cancelled"`
+		Layers      map[int]LayerStatus `json:"layers"`
+	}
+
+	tee struct {
```

Comment:
Move everything tee and its methods in a different file

Reply:
Done